recipes/beckhoff-ads/recipe.yaml

```
@@ -0,0 +1,52 @@
+context:
+  name: beckhoff-ads
+  version: 22
+
+package:
+  name: ${{ name|lower }}
+  version: ${{ version }}.0.0
+
+source:
+  url: https://github.com/Beckhoff/ADS/archive/refs/tags/v${{ version }}.tar.gz
+  sha256: 7330e032a7add1f97df6759cc1610c06760ae928ecfa4531915a3595ab70aaaf
+  patches: 
+    # meson.build on v22 does not provide install directives - add them via patch. To be checked for newer versions.
+    - meson-build.patch
+
+build:
+  number: 0
+  skip:
+    - win
+  script: |
+    meson setup build --prefix="$PREFIX" -Dlibdir="$PREFIX/lib" -Dwerror=false
+    ninja -C build install
+
+requirements:
+  build:
+    - ${{ stdlib("c") }}
+    - ${{ compiler('cxx') }}
+    - meson
+    - ninja
+
+tests:
+  - script:
+      - test -d "${PREFIX}/include/AdsLib"
+      - test -f "${PREFIX}/include/AdsLib/AdsLib.h"
+      - test -f "${PREFIX}/bin/adstool"
```

This [should be a functional CLI](https://github.com/Beckhoff/ADS/blob/81a402053a19365ba546ec0dc29331b245f9eb71/AdsTool/main.cpp#L47), so how about testing execution instead of existence?

```suggestion
      - adstool --help
```

---

Agreed!